BDArmory/Weapons/BDExplosivePart.cs

```diff
@@ -324,11 +323,8 @@ public void DetonateIfPossible()
             if (!HighLogic.LoadedSceneIsFlight || part == null) return;
             if (!hasDetonated && Armed)
             {
-                if (warheadType == "standard")
-                {
-                    //direction = (part.transform.position + part.rb.velocity * Time.deltaTime).normalized;
-                    direction = default(Vector3);
-                }
+				direction = part.partTransform.forward; //both the missileReferenceTransform and smallWarhead part's forward direction is Z+, or transform.forward.
+                // could also do warheadType == "standard" ? default: part.partTransform.forward, as this simplifies the isAngleAllowed check in ExplosionFX, but at the cost of standard heads always being 360deg blasts (but we don't have limited angle balsts for missiels at present anyway, so not a bit deal RN)
```

Comment:
Doesn't not using `default` for standard warheads change the default for them from 360° to 100° cone (`eFx.AngleOfEffect` being set on line 1025 of ExplosionFX.cs)?

Also, looking at where `ExplosionFX.CreateExplosion` is called from, is flak supposed to be directional?

Reply:
Hmm. Yeah, it would. Do we want standard warheads to be angle limited to a 100° cone?
For the flak question - is this from rockets with the flak tag? I guess they could be, the flak tag was just being used for proxi detonation.

Reply:
No idea if 100° cone or the full 360° is better as the default for standard warheads. Maybe ask Josue and the others that use them more what the default should be?

The flak I was meaning is from line 431 of PooledBullet.cs. It's using `currentVelocity` for the direction. I had the impression flak was more along the lines of "lob up a shell, it explodes with shrapnel in all directions", but my knowledge of weapons is fairly limited.

Reply:
Should probably be full 360 for standard warheads, mostly for the sake of bombs. I half-want it to be a cone because that's what the damage would look like from a missile or other fast moving projectile exploding since they've got that forwards component in terms of velocity and would throw shrapnel etc. forwards but if we do that it'd mess up how bombs work and our damage is mostly from blasts.

Reply:
That's only the default if the `direction` vector is not `default`, which it is for most explosions.

Reply:
With the latest commits below and 513f775 in dev, I think this is ready to merge.
Is there any other reloadable/multi-missile stuff that you're working on @SuicidalInsanity that I should wait for?

Reply:
@BrettRyland 
No, this is it for the moment. The remaining stuff that's still pending (customizable VLS implementation) can wait for now until I get it working fully, no sense in having that hold things up when the rest of it is fully implemented and works. We can go ahead and merge.